File: src/main/kotlin/com/github/darderion/mundaneassignmentpolice/checker/RuleViolation.kt
package com.github.darderion.mundaneassignmentpolice.checker

import com.github.darderion.mundaneassignmentpolice.pdfdocument.text.Line

enum class RuleViolationType {
	Система, Ошибка, Предупреждение
}

data class RuleViolation(
	val lines: List<Line>,
	val message: String,
	val type: RuleViolationType
) {
	// override fun toString() = if (lines.count() == 1) "[${lines.first().line}, p.${lines.first().page}] --> '$message'" else ""
}

File: src/main/webapp/src/classes/RuleViolation.ts
import PDFLine from "./PDFLine";

export default class RuleViolation {
	lines: PDFLine[]
	message: string;
	type: string;

	constructor(lines: PDFLine[], message: string, type: string) {
		this.lines = lines
		this.message = message
		this.type = type
	}
}
